homeassistant/components/freebox/router.py

```diff
@@ -0,0 +1,228 @@
+
+        # Devices & sensors
+        await self.update_all()
+        async_track_time_interval(self.hass, self.update_all, SCAN_INTERVAL)
+
+    async def update_all(self, now: Optional[datetime] = None) -> None:
+        """Update all Freebox platforms."""
+        await self.update_sensors()
+        await self.update_devices()
+
+    async def update_devices(self) -> None:
+        """Update Freebox devices."""
+        new_device = False
+        fbx_devices: Dict[str, any] = await self._api.lan.get_hosts_list()
+
+        # Adds the Freebox itself
+        fbx_devices.append(
+            {
+                "primary_name": self.name,
+                "l2ident": {"id": self.mac},
+                "vendor_name": self.manufacturer,
+                "host_type": "router",
+                "active": True,
+                "attrs": self._attrs,
+            }
+        )
+
+        for fbx_device in fbx_devices:
+            device_mac = fbx_device["l2ident"]["id"]
+
+            if self._devices.get(device_mac) is None:
+                new_device = True
+
+            self._devices[device_mac] = fbx_device
+
+        async_dispatcher_send(self.hass, self.signal_device_update)
+
+        if new_device:
+            async_dispatcher_send(self.hass, self.signal_device_new)
+
+    async def update_sensors(self) -> None:
+        """Update Freebox sensors."""
+        # System sensors
+        syst_datas: Dict[str, any] = await self._api.system.get_config()
+        temperature_datas = {item["id"]: item for item in syst_datas["sensors"]}
+        # According to the doc it is only temperature sensors in celsius degree.
+        # Name and id of the sensors may vary under Freebox devices.
+
+        for sensor_key, sensor_attrs in temperature_datas.items():
+            self._temperature_sensors[sensor_key] = sensor_attrs["value"]
+
+        # Connection sensors
+        connection_datas: Dict[str, any] = await self._api.connection.get_status()
+        for sensor_key, sensor_attrs in CONNECTION_SENSORS.items():
+            self._connection_sensors[sensor_key] = connection_datas[sensor_key]
+
+        self._attrs = {
+            "IPv4": connection_datas.get("ipv4"),
+            "IPv6": connection_datas.get("ipv6"),
+            "connection_type": connection_datas["media"],
+            "uptime": datetime.fromtimestamp(
+                round(datetime.now().timestamp()) - syst_datas["uptime_val"]
+            ),
+            "firmware_version": self.firmware_version,
+            "serial": syst_datas["serial"],
+        }
+
+        async_dispatcher_send(self.hass, self.signal_sensor_update)
+
+    async def reboot(self) -> None:
+        """Reboot the Freebox."""
+        await self._api.system.reboot()
+
+    async def close(self) -> None:
+        """Close the connection."""
+        if self._api is not None:
+            await self._api.close()
+        self._api = None
+
+    @property
+    def name(self) -> str:
```

Comment:
Please remove all the properties that just return an instance attribute. If we don't need to do any more modifications in the property we can just use a regular instance attribute.